src/searchee.ts

```diff
@@ -95,19 +104,41 @@ export function getLargestFile(files: File[]): File {
 	return files.reduce((a, b) => (a.length > b.length ? a : b));
 }
 
-export async function getNewestFileAge(searchee: Searchee): Promise<number> {
-	if (searchee.infoHash || searchee.path) {
-		throw new Error("Only virtual searchees have absolute paths");
-	}
+export function getAbsoluteFilePath(
+	sourceRoot: string,
+	filePath: string,
+	sourceRootIsFileHint?: boolean,
+): string {
+	return sourceRootIsFileHint ?? statSync(sourceRoot).isFile()
+		? sourceRoot
+		: join(dirname(sourceRoot), filePath);
+}
+
+export async function getNewestFileAge(
+	absoluteFilePaths: string[],
+): Promise<number> {
 	return Math.max(
 		...(await Promise.all(
-			searchee.files.map((file) =>
-				stat(file.path).then((s) => s.mtimeMs),
-			),
+			absoluteFilePaths.map((file) => stat(file).then((s) => s.mtimeMs)),
 		)),
 	);
 }
 
+export async function getSearcheeNewestFileAge(
+	searchee: SearcheeWithoutInfoHash,
+): Promise<number> {
+	if (!searchee.path) {
+		return getNewestFileAge(searchee.files.map((file) => file.path));
+	}
+	const pathStat = statSync(searchee.path);
+	if (pathStat.isFile()) return pathStat.mtimeMs;
+	return getNewestFileAge(
+		searchee.files.map((file) =>
+			getAbsoluteFilePath(searchee.path!, file.path, false),
```

Comment:
TS would ordinarily narrow this to non-undefined automatically, but I think it's worried that mutation could remove `path` before the lambda gets called. 

You should be able to trick it into understanding that searchee.path is defined by rebinding `path` above the `.map` call with `const { path } = searchee;`

Reply:
I prefer using ! as it a no-op.

Reply:
Using ! puts the onus on the programmer to validate that they know better than the computer. We should rely on the computer wherever we can, and a variable binding is basically 0 cost.